### PR TITLE
[WIP]  implementation of RDD model method

### LINKED ISSUE #22: implement track structure methods

### PR DESCRIPTION
**Pull request by grzanka**
Fixes #22

### CODE REVIEW

File: src/functionsFromC/AT_D_RDD_Gy.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+	/*********************STANDARD PARAMETER*************************/
+	if(!parameters.stopping_power_source_no){
+		 alert("MESSAGE TO DEVELOPER: NO PARAMETER stopping_power_source_no IN OBJECT PASSED TO THIS FUNCTIONS");
+		 return "error";
+	}
+	let stopping_power_source_no = parameters.stopping_power_source_no;
+
+	/*********************INPUT ARRAY********************************/
+	if(!parameters.D_RDD_Gy){
```
Comment (grzanka) [inline src/functionsFromC/AT_D_RDD_Gy.js]: - [ ] Why D_RDD_Gy hasn't been translated to "OUTPUT ARRAY" @adudzik @marwin1991  ? It is autogenerated from:
```
int AT_D_RDD_Gy( const long  n,
    const double  r_m[],
    const double  E_MeV_u,
    const long    particle_no,
    const long    material_no,
    const long    rdd_model,
    const double  rdd_parameter[],
    const long    er_model,
    const long    stopping_power_source_no,
    double        D_RDD_Gy[])
```
image

**Comment by grzanka** [comment]
@adudzik @marwin1991 this PR is just for testing purposes - I'd like to make some exercise in adding functionalities myself. Please take a look here and comment what I am doing wrong

**Comment by marwin1991** [comment]
To tell ctojsconverter that param od used to return value you need to add at the end od param "@length". In this case it should be `double D_RDD_Gy[]@n`

**Comment by marwin1991** [comment]
Its described when you hit show help button or show example. I will add issue to describe usage in docs

File: src/static/json/RDD/RDD.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      "type": "select",
+      "parameterName": "rdd_model",
+      "label": "RDD model",
+      "values": "rdd_models",
+      "asManyAsPoints": false,
+      "defaultValue": 2,
+      "description": "Choose RDD model"
+    },
+    {
+      "type": "select",
```
Comment (grzanka) [inline src/static/json/RDD/RDD.json]: @marwin1991 @adudzik rdd_parameters store the array of parameters. It is fixed size array of 3 elements. Is it possible to use such data types here ?

Comment (marwin1991) [reply src/static/json/RDD/RDD.json]: First it needs to add dictionary. Then use it. Same as materials or particles.
https://libamtrackweb.readthedocs.io/en/latest/jsons.html#gcjson

Comment (grzanka) [reply src/static/json/RDD/RDD.json]: But I'd like to provide the values (up to 3 numbers) in the input text fields in the web interface. If not possible I'll add another C wrapper to reduce the number of RDD parameters to a single one.

Comment (adudzik) [reply src/static/json/RDD/RDD.json]: I'm not sure if I understood you correctly but it's not possible (in currently supported input types) to insert more than one value in single input.  It's not necessary to create another wrapper, we can extend our form items with e.g. multiselect or something like that.

Comment (grzanka) [reply src/static/json/RDD/RDD.json]: RDD models needs extra parameters which are put together into single array. Some models (i.e. Geiss) require a single number which goes into rdd_parameter[0], rest is ignored. Other models require 2 or 3 numbers.

**Comment by marwin1991** [comment]
#26 will contain changes from this pr beacause dictionary added here also in #26 are used

**Pull request merged**